frontend/homework/14/src/homepage/body/Body.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { Link, useParams } from 'react-router-dom';
+import { ItemsContext } from '../../App';
+import { FilteredItemsContext, SearchItemsContext, SortItems, SortItemsContext } from '../MainHome';
+import { Item } from '../../interfaces/Item';
+
+export function Body() {
+    const { items, setItems } = useContext(ItemsContext);
+    useEffect(() => {
+        fetchItems();
+    }, []);
+    const [filteredItems, setFilteredItems] = useState<Item[]>([]);
+    const { searchedItem, setSearchedItem } = useContext(SearchItemsContext);
+    const { selectedFilter, setSelectedFilter } = useContext(FilteredItemsContext);
+    const { selectedSortOrder, setSelectedSortOrder } = useContext(SortItemsContext);
+    const { category } = useParams<{ category: string }>();
+
+    useEffect(() => {
+        let filteredByCategory = category ? items.filter(item => item.category === category) : items;
+        filteredByCategory = selectedFilter
+            ? filteredByCategory.filter(item => item.category === selectedFilter)
+            : filteredByCategory;
+        const filteredBySearch = searchedItem
+            ? filteredByCategory.filter(item =>
+                item.title.toLowerCase().includes(searchedItem.toLowerCase())
+            )
+            : filteredByCategory;
+
+        const sortedItems = selectedSortOrder === SortItems.Ascending
+            ? [...filteredBySearch].sort((a, b) => a.price - b.price)
+            : [...filteredBySearch].sort((a, b) => b.price - a.price);
+
+        setFilteredItems(sortedItems);
+    }, [selectedFilter, searchedItem, items, selectedSortOrder]);
```

Comment:
this section could be repetitive. Ideally Lets use a custom hook for this logic.